frontend/pages/ShoppingPage.js

```diff
@@ -103,7 +103,7 @@ export default class ShoppingPage extends React.Component {
                         return(
```

Comment:
Starting on line 100, is it not possible to move this giant blob into its own component? It might reduce clutter and help with readability. So instead of mapping your list on this giant inline object, that inline object could be its own component that you pass in params as necessary into.

Reply:
Noted. Added issue #128